### PR TITLE
Add support for entity_view

### PR DESCRIPTION
**Pull request by steveworley**
`entity_view` was [deprecated in 8.0.0](https://api.drupal.org/api/drupal/core%21includes%21entity.inc/function/entity_view/8.8.x).

Some complexities with the change in API as moving to this no longer supports the `$reset` parameter, but this will convert `entity_view` to a Drupal 9 compatible expression.

### CODE REVIEW

File: rector_examples/entity_view.php
Comment (damontgomery) [inline rector_examples/entity_view.php]: minor: spelling of `$view`.

File: rector_examples/entity_view.php
Comment (damontgomery) [inline rector_examples/entity_view.php:25]: Thanks for calling this out.

File: src/Rector/Deprecation/Base/EntityViewBase.php
Comment (damontgomery) [inline src/Rector/Deprecation/Base/EntityViewBase.php]: Swap this for `$entityTypeManager_method_name`? like the ones below?

File: src/Rector/Deprecation/Base/EntityViewBase.php
Comment (damontgomery) [inline src/Rector/Deprecation/Base/EntityViewBase.php]: Could we make this `$entity_reference`? I like how the names you've used are mostly matching the method names like `$getViewBuilder_method_name` which is very clear to me. :)

File: src/Rector/Deprecation/Base/EntityViewBase.php
Comment (damontgomery) [inline src/Rector/Deprecation/Base/EntityViewBase.php]: Can we combine `EntityViewBase.php` and `EntityViewRector`? It looks like there was a base class, which we could refactor later, but for now, putting them together makes more sense to me while reading. The comments seem like they need to be updated too.

File: src/Rector/Deprecation/Base/EntityViewBase.php
Comment (damontgomery) [inline src/Rector/Deprecation/Base/EntityViewBase.php]: Did you want to add a comment to the replacement about checking for the `reset` option?

If not, we could remove this trait.

**Review by damontgomery — CHANGES_REQUESTED** [review]
Thanks, @steveworley! And, hello!

I left a few comments. This looks close.

There are three other requirements for adding new rules
- Adding the new rector rule to the configuration file, config/drupal-8/drupal-8.0-deprecations.yml
- Updating the index (deprecation-index.yml)
- Adding the after example like rector_examples_updated/entity_create.php.

Also, there is a fairly large refactor going on to support newer versions of Rector. Because of that, I'm hesitant to merge new rules over the next week or so. If you want to build off that branch, we're in the final stages of moving in that direction. #134. This code should mostly work, but the configuration files have moved from YAML to php and some of the `use` statements have changed. Example: https://github.com/palantirnet/drupal-rector/pull/134/files#diff-069c02b1633ca197bab8d62713679cae3a5a80118c3b0a4ee51100c749ae6360.

Sorry again for the pieces moving from under you and happy to help take a look or make the updates after that PR is merged if you don't want to or don't have the time to make the updates.

**Comment by steveworley** [comment]
Thanks for the review @damontgomery! Fancy seeing you here :)

I updated all the points and rebased against the updated rector branch let me know if you need anything else changed.

**Comment by damontgomery** [comment]
:wave: I think reasonably I can take a look next week.

Thanks for the updates!

File: src/Rector/Deprecation/EntityViewRector.php
Comment (damontgomery) [inline src/Rector/Deprecation/EntityViewRector.php:66]: minor: spelling of `$entityTypeManager` missing the `e` in `Type`.

**Review by damontgomery — APPROVED** [review]
Thanks for the updates. The code is clear! The automated tests pass.

I would be happy to have this merged. Not sure if we should wait until / if we merge `rector-0.10-minimal-changes`.

**Pull request merged**
